### PR TITLE
test: Add Completed Hollow Account fuzzy tests

### LINKED ISSUE #11677: Add fuzzing suite for operations on completed hollow accounts
Currently we have two fuzzing suites related to hollow accounts:

1. `AddressAliasIdFuzzing` which tests creation of hollow accounts
2. `HollowAccountCompletionFuzzing` which tests the completion of hollow accounts
 
Additionally we want to have fuzzing suite for verifying that completed hollow accounts can participate in any kind of transaction since they should be fully-fledged accounts.

### PR DESCRIPTION
**Pull request by 0xivanov**
**Description**:

Currently we have two fuzzing suites related to hollow accounts:

1. `AddressAliasIdFuzzing` which tests creation of hollow accounts
2. `HollowAccountCompletionFuzzing` which tests the completion of hollow accounts
 
Additionally we want to have fuzzing suite for verifying that completed hollow accounts can participate in any kind of transaction since they should be fully-fledged accounts.

This PR adds fuzzing tests for completed hollow account and fixes `UmbrellaRedux` as it was not working as expected.

**Related issue(s)**:

Fixes #11677

**Notes for reviewer**:

**Checklist**

- [ ] Documented (Code comments, README, etc.)
- [ ] Tested (unit, integration, etc.)

### CODE REVIEW

File: hedera-node/test-clients/src/main/java/com/hedera/services/bdd/spec/infrastructure/selectors/HollowAccountSelector.java [hunk 1]
```diff
@@ -0,0 +1,34 @@
...
+ * limitations under the License.
+ */
+
+package com.hedera.services.bdd.spec.infrastructure.selectors;
+
+import java.util.Optional;
+import java.util.Set;
+import java.util.function.BiFunction;
+
+public class HollowAccountSelector implements BiFunction<Set<String>, Set<String>, Optional<String>> {
```
Comment (0xivanov) [inline hedera-node/test-clients/src/main/java/com/hedera/services/bdd/spec/infrastructure/selectors/HollowAccountSelector.java:23]: Remove this class

File: hedera-node/test-clients/src/main/java/com/hedera/services/bdd/suites/regression/factories/HollowAccountCompletedFuzzingFactory.java [hunk 2]
```diff
@@ -0,0 +1,259 @@
...
+     */
+    public static HapiSpecOperation[] initOperations() {
+        return new HapiSpecOperation[] {
+            cryptoCreate(LAZY_CREATE_SPONSOR)
+                    .balance(UNIQUE_PAYER_ACCOUNT_INITIAL_BALANCE)
+                    .withRecharging(),
+            cryptoCreate(CRYPTO_TRANSFER_RECEIVER)
+                    .balance(UNIQUE_PAYER_ACCOUNT_INITIAL_BALANCE)
+                    .withRecharging(),
+            cryptoCreate(UNIQUE_PAYER_ACCOUNT)
```
Comment (0xivanov) [inline hedera-node/test-clients/src/main/java/com/hedera/services/bdd/suites/regression/factories/HollowAccountCompletedFuzzingFactory.java:110]: Looks like this is not used

File: hedera-node/test-clients/src/main/java/com/hedera/services/bdd/suites/regression/factories/HollowAccountCompletedFuzzingFactory.java [hunk 3]
```diff
@@ -0,0 +1,259 @@
...
+    public static Function<HapiSpec, OpProvider> hollowAccountFuzzingWith(final String resource) {
+        return spec -> {
+            final var props = RegressionProviderFactory.propsFrom(resource);
+
+            final var accounts = new RegistrySourcedNameProvider<>(
+                    AccountID.class, spec.registry(), new HollowAccountSelector(HOLLOW_ACCOUNT));
+            final var tokens = new RegistrySourcedNameProvider<>(TokenID.class, spec.registry(), new RandomSelector());
+            var tokenRels = new RegistrySourcedNameProvider<>(
+                    TokenAccountRegistryRel.class, spec.registry(), new RandomSelector());
+            // TODO new RandomSelector(account -> !account.equals("kvoto si iskam")));
```
Comment (0xivanov) [inline hedera-node/test-clients/src/main/java/com/hedera/services/bdd/suites/regression/factories/HollowAccountCompletedFuzzingFactory.java:128]: TODO

File: hedera-node/test-clients/src/main/java/com/hedera/services/bdd/suites/regression/factories/RegressionProviderFactory.java [hunk 4]
```diff
@@ -141,11 +139,11 @@ public static Function<HapiSpec, OpProvider> factoryFrom(Supplier<String> resour
                                             RandomTransfer.DEFAULT_RECORD_PROBABILITY,
                                             props)),
                             intPropOrElse("randomTransfer.bias", 0, props))
+                    //                    .withOp(
```
Comment (0xivanov) [inline hedera-node/test-clients/src/main/java/com/hedera/services/bdd/suites/regression/factories/RegressionProviderFactory.java:142]: TODO

File: hedera-node/test-clients/src/main/java/com/hedera/services/bdd/spec/infrastructure/OpProvider.java [hunk 5]
```diff
@@ -44,15 +43,14 @@ public interface OpProvider {
...
-        INVALID_SIGNATURE,
         DUPLICATE_TRANSACTION,
         INVALID_PAYER_SIGNATURE,
         INSUFFICIENT_PAYER_BALANCE,
         PLATFORM_TRANSACTION_NOT_CREATED
     };
 
     ResponseCodeEnum[] STANDARD_PERMISSIBLE_OUTCOMES = {
-        SUCCESS, LIVE_HASH_NOT_FOUND, INVALID_SIGNATURE, INSUFFICIENT_PAYER_BALANCE, UNKNOWN
+        SUCCESS, LIVE_HASH_NOT_FOUND, INSUFFICIENT_PAYER_BALANCE, UNKNOWN
```
Comment (0xivanov) [inline hedera-node/test-clients/src/main/java/com/hedera/services/bdd/spec/infrastructure/OpProvider.java:53]: Add timeout for the BiasProvider

File: hedera-node/test-clients/src/main/java/com/hedera/services/bdd/spec/infrastructure/OpProvider.java [hunk 6]
```diff
@@ -44,15 +43,14 @@ public interface OpProvider {
...
         BUSY,
-        INVALID_SIGNATURE,
         DUPLICATE_TRANSACTION,
         INVALID_PAYER_SIGNATURE,
         INSUFFICIENT_PAYER_BALANCE,
         PLATFORM_TRANSACTION_NOT_CREATED
     };
 
     ResponseCodeEnum[] STANDARD_PERMISSIBLE_OUTCOMES = {
-        SUCCESS, LIVE_HASH_NOT_FOUND, INVALID_SIGNATURE, INSUFFICIENT_PAYER_BALANCE, UNKNOWN
```
Comment (kselveliev) [inline hedera-node/test-clients/src/main/java/com/hedera/services/bdd/spec/infrastructure/OpProvider.java:55]: Probably we should not remove INVALID_SIGNATURE from the standard prechecks and outcomes

Comment (0xivanov) [reply hedera-node/test-clients/src/main/java/com/hedera/services/bdd/spec/infrastructure/OpProvider.java:55]: There are 50/50 cases where we don't want this status/where we want it, so i've added `customOutcomes`, which contains `INVALID_SIGNATURE` to the operations that require it.
This way it's more future proof.

File: hedera-node/test-clients/src/main/java/com/hedera/services/bdd/suites/regression/CompletedHollowAccountOperationsFuzzing.java [hunk 7]
```diff
@@ -53,7 +53,7 @@ final HapiSpec completedHollowAccountOperationsFuzzing() {
         return defaultHapiSpec("CompletedHollowAccountOperationsFuzzing")
                 .given(initOperations())
                 .when()
-                .then(runWithProvider(hollowAccountFuzzingWith(PROPERTIES)).lasting(10L, TimeUnit.SECONDS));
+                .then(runWithProvider(hollowAccountFuzzingWith(PROPERTIES)).lasting(60L, TimeUnit.MINUTES));
```
Comment (kselveliev) [inline hedera-node/test-clients/src/main/java/com/hedera/services/bdd/suites/regression/CompletedHollowAccountOperationsFuzzing.java:56]: Revert to seconds

**Review by kselveliev — APPROVED** [review]
LGTM

**Review by tinker-michaelj — APPROVED** [review]
LGTM, tyvm @0xivanov !

**Review by kselveliev — DISMISSED** [review]

File: hedera-node/test-clients/src/main/java/com/hedera/services/bdd/spec/infrastructure/providers/ops/contract/RandomContract.java [hunk 8]
```diff
@@ -88,6 +88,8 @@ public Optional<HapiSpecOperation> get() {
         final SupportedContract choice = choices[n % choices.length];
 
         HapiContractCreate op = contractCreate(tentativeContract)
+                .payingWith(key.get())
```
Comment (tinker-michaelj) [inline hedera-node/test-clients/src/main/java/com/hedera/services/bdd/spec/infrastructure/providers/ops/contract/RandomContract.java:91]: Does this mean all payer accounts used with this `RandomContract` provider must be auto-created accounts?

Unclear how this works.

Comment (0xivanov) [reply hedera-node/test-clients/src/main/java/com/hedera/services/bdd/spec/infrastructure/providers/ops/contract/RandomContract.java:91]: Not necessarily, the user of this class can specify the initiator account of the `contract create transaction`, which can be auto-created, normal account, admin account 0.0.2, etc.

Comment (0xivanov) [reply hedera-node/test-clients/src/main/java/com/hedera/services/bdd/spec/infrastructure/providers/ops/contract/RandomContract.java:91]: The point of the tests is to check if auto-created account can sign and pay for such transactions, so i've added this option. Beforehand it was using the default payer and singer (account 0.0.2 of I'm not mistaken)

**Review by kselveliev — APPROVED** [review]

File: hedera-node/test-clients/src/main/java/com/hedera/services/bdd/spec/infrastructure/providers/ops/contract/RandomContract.java (hunk 8 above)
Comment (tinker-michaelj) [reply hedera-node/test-clients/src/main/java/com/hedera/services/bdd/spec/infrastructure/providers/ops/contract/RandomContract.java:91]: Hmm well if it works, it works! 💯

**Review by tinker-michaelj — APPROVED** [review]
LGTM, ty @0xivanov !

**Pull request merged**